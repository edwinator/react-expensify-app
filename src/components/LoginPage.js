import React from "react";
import {connect} from "react-redux";
import {startLoginGoogle, startLoginFB} from "../actions/auth";

export const LoginPage = ({startLoginGoogle, startLoginFB}) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>Get expenses under control</p>
      <button className="button" onClick={startLoginGoogle}>
        Login with Google
      </button>
      <button className="button button--disabled" onClick={startLoginFB}>
        Login with Facebook
      </button>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLoginGoogle: () => dispatch(startLoginGoogle()),
  startLoginFB: () => dispatch(startLoginFB()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
