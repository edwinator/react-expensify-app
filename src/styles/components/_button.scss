.button {
  background: $light-blue;
  border: none;
  text-align: center;
  padding: $s-size;
  font-size: $font-size-l;
  font-weight: 300;
  color: white;
  display: inline-block;
  text-decoration: none;
  line-height: 1;
  margin-top: 0.7rem;
}

.button--link {
  background: none;
}
