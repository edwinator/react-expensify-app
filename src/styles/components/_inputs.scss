.text-input {
  border: 1px solid #cacccd;
  height: 50px;
  padding: $s-size;
  font-size: $font-size-l;
  font-weight: 300;
}

.select {
  @extend .text-input;
}

.test-area {
  @extend .text-input;
  height: 10rem;
}