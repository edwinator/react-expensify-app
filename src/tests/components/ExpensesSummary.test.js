import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';
// import expenses from '../fixtures/expenses';

test('should render ExpensesSummary with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={20}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with two expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={5} expensesTotal={278978970}/>);
  expect(wrapper).toMatchSnapshot();
});

// test('return 0 if no expenses', () => {
//     const result = selectExpensesTotal([]);
//     expect(result).toEqual(0);    
// });

// test('return single expense amount', () => {
//     const result = selectExpensesTotal([expenses[0]]);
//     expect(result).toEqual(expenses[0].amount);    
// });

// test('return total from multiple expenses', () => {
//     const result = selectExpensesTotal(expenses);
//     expect(result).toEqual(expenses[0].amount + expenses[1].amount + expenses[2].amount);    
// });