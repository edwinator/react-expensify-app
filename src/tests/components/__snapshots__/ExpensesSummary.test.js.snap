// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $0.20
  </h1>
</div>
`;

exports[`should render ExpensesSummary with two expenses 1`] = `
<div>
  <h1>
    Viewing 
    5
     
    expenses
     totalling 
    $2,789,789.70
  </h1>
</div>
`;
